first pass readme update for fluid-react-starter (#250)
microsoft/FluidExamples: fluid-react-starter/README.md

npm run start
```

## Modifying the model

### Specify additional DDSes

> Inside of `src/config.ts`, in the `containerConfig` you can modify the `initialObjects` that are returned by the container. 

Reviewer: ```suggestion
Inside of `src/config.ts`, you can define the `initialObjects` that are returned by the container in the `containerSchema`. 
```

Reviewer: Also, this variable itself in config.ts should also be renamed `containerSchema` to match the parameter name and interface

To add another DSS to this list, make sure that the DDS is imported from `@fluid-experimental/fluid-framework`, select a key, and add the DDS to `initialObjects`.

```ts
import { SharedMap, SharedCounter } from '@fluid-experimental/fluid-framework';

> export const containerConfig = {

Reviewer: ```suggestion
export const containerSchema = {
```
The config is the first param and the schema is the second in create/getContainer calls

  name: 'cra-demo-container',
  initialObjects: {
    myMap: SharedMap,
    myCounter: SharedCounter
  },
};
```
### Update the `defaultData` of those DDSes

Inside of `src/config.ts` you can modify the `setDefaultData` function to change the data added to the initial DDSes upon container creation. Any `initialObjects` specified above will be available on `fluidContainer.initialObjects`. 

```ts
export const setDefaultData = (fluidContainer: FluidContainer) => {
  const { myMap, myCounter } = fluidContainer.intitialObjects;
  myCounter.increment(1); // start at 1
}
```

### Update the `model` to access and modify your Fluid data

> All of our application's business logic will be stored in `model.ts`, which is a class with access to both the `FluidContainer` and the `FrsContainerServices`. 

Reviewer: ```suggestion
All of your application's business logic will be stored in `model.ts`, which is a class with access to both the `FluidContainer` and the `FrsContainerServices`. 
```

> In the `FluidModel` class we can expose new properties and methods that can leverage any of the default DDSes or `FrsContainerServices` properties, like `audience`. These properties and methods will be used to provide a `store` of data and actions for our view to access, so keep the model focused on lower level access of the data itself.

Reviewer: ```suggestion
In the `FluidModel` class you can expose new properties and methods that can leverage any of the DDSes provided by your `FluidContainer` or `FrsContainerServices` properties, like `audience`. These properties and methods will be used to provide a `store` of data and actions for your view to access, so keep the model focused on lower level access of the data itself.
```

```ts
  // inside of the constructor
    this.counter = container.initialObjects.myCounter;
  //
  public getCounterValue = () => {
    return this.counter.value;
  }
  public jumpCounterFive = (): number => {
    return this.counter.increment(5)
  }
```

### Write custom events

> This demo is written to funnel all model events through the `modelChanged` event. Each event emit accepts a payload that can used to differentiate one even from another. 

Reviewer: ```suggestion
This template is written to funnel all model events through the `modelChanged` event. Each event emit accepts a payload that can used to differentiate one event from another. 
```

> 

Reviewer: May want to include a bit of context, i.e. this is an example for using a SharedCounter and its `incremented` event -> link to SharedCounter docs

### Create a store

// TODO: split this out into 4 parts and walk through creation

```ts
export const useGetCounterStore = () => useGetStore({
  initialState: model => model.getCounterValue(),
  queries: {
    getCounter: state => state,
    isMoreThan100: state => state > 100
  },
  actions: {
    increment: ( model, payload: {double: boolean} ) => {
      if (double) { 
        model.jumpCounterFive(); 
        model.jumpCounterFive(); 
      } else {
        model.jumpCounterFive();
      }
    }
  },
  reducer: (model, draft, { type, data }) => {
    switch (type) {
      case "counterIncremented":
        draft = data.newValue
        break;          
    }
  },
})
```
   

### Import and use the store

// TODO: walkthrough store usage

```tsx
const CounterPage = (props) => {
  const {
    dispatch,
    actions: { increment },
    queries: { getCounter, isMoreThan100 }
  } = useGetCounterStore();

  return(
    <div> 
    Counter is {isMoreThan100 ? "TOO BIG" : "kinda small" }  </br>
    <button onClick={() => dispatch(increment())} > jump </button> </br>
    <button onClick={() => dispatch(increment(double: true))} > BIG jump! </button> </br>
    </div>
  )
}

```

> 

Reviewer: nit: newlines, but perhaps you're leaving them here as spacing for future content to fill out?

    };
```

> When just starting the app with `npm run start`, the `useFrs` value here is false and the second set of values will be used. Here, we see that our orderer and storage URLs that point to the service are directed towards 'http://localhost:7070'. The `user` object being passed into the `InsecureTokenProvider` will identify the current member's user ID and user name in the application.

Reviewer: Since we're splitting the local/FRS sections apart, could we reword this paragraph and move it up to "Run the app locally" so readers know whats happening when just `npm run start` is used?

Author: My goal here was to punt on the complexity of frs vs local configs, as the user doesn't need to even read those to start the project. Once we get down to FRS connection we talk about swapping configs due to the boolean.
